### PR TITLE
Added Chapter 2 Features: The Mercy System + chapter2 variable

### PR DESCRIPTION
**Pull request by Simbel0**
The new mercy system introduced in Deltarune Chapter 2 recreated in CYK!

This system includes with it the new `chapter2` variable, which will tell the engine if he should use the new features of Chapter 2 or not.
And the pull request also comes with a new Example file (with the new Poseur script that goes with it) that changes 3 or 4 things from the default example battle to showcase how the new functions works.

### CODE REVIEW

File: Lua/Libraries/CYK/CYKCore.lua [hunk 1]
```diff
@@ -949,9 +958,29 @@ return function ()
             -- Play the EndBattle animation of each Players and display the end text
             for i = 1, #self.players do
                 self.SetAnim(self.players[i], "EndBattle")
+                if chapter2 then
+                    if self.stronger then
+                        local hpAdd={2, 2, 2}
```
Comment (RhenaudTheLukark) [inline Lua/Libraries/CYK/CYKCore.lua]: I don't get this code. Why do you make a table and try to store the players' hpAdd values like that? You don't use them afterwards.

File: Lua/Libraries/CYK/Sandboxing/Entity.lua [hunk 2]
```diff
@@ -258,6 +259,84 @@ function AddSpell(name, description, tpCost, targetType)
...
+        else
+            CYK.UI.CreateChangeText(val, target, nil, true)
+        end
+    end
+end
+
+-- Returns the percentage of mercy of an enemy. Returns -1 if the enemy can't be spared.
+function GetMercyPercent(of)
+    if not chapter2 then
+        error("entity.GetMercyPercent() is a function only usable in a mod with the functionnabilities of Chapter 2!\n\nSet chapter2 to true in the Encounter file to access it.")
```
Comment (RhenaudTheLukark) [inline Lua/Libraries/CYK/Sandboxing/Entity.lua]: I feel like I read that word differently every time I see it. "Functionnalities" or "Functionnalities" can easily be replaced by "Features".

**Review by RhenaudTheLukark — CHANGES_REQUESTED** [review]
I combed the code and it looks solid! Just remember that if you check a boolean and said boolean variable is nil, then it'll be considered as false, so no need to set it to false.

File: Lua/Libraries/CYK/Sandboxing/Entity.lua (hunk 2 above)
Comment (Simbel0) [reply Lua/Libraries/CYK/Sandboxing/Entity.lua]: You're right
I fixed it, it's all good ~~and without any ugly typos~~ now!

File: Lua/Libraries/CYK/CYKCore.lua (hunk 1 above)
Comment (RhenaudTheLukark) [reply Lua/Libraries/CYK/CYKCore.lua]: You didn't look into this comment!

Comment (Simbel0) [reply Lua/Libraries/CYK/CYKCore.lua]: I did
I just fix the easiest part first

I actually don't get it either now that I look at it again

I think what I was planning to do was to create the table of HPs to add to the players' `hp` and `maxhp` values (each value in the table would be for the corresponding player) and then apply it to said player but it kind of got a dirty mess
I'm going to rewrite that asap

Comment (Simbel0) [reply Lua/Libraries/CYK/CYKCore.lua]: I finished remaking that part
It should now work AND make sense

**Comment by RhenaudTheLukark** [comment]
All seems to be good now! You talked about updating the docs, so if you wanna do that, it'd be greatly appreciated!

**Comment by Simbel0** [comment]
I have finished updating the documentation too
I didn't show every changes I made on Discord so I let you check if everything's okay to you

File: Documentation CYK/documentation.html [hunk 3]
```diff
@@ -107,7 +107,7 @@ <h3><span class="CYK"></span> List of sections with new/changed content in CYK:<
                 <br><br>
 
                 <i><u>Misc. Functions</u></i><br>
-                The function <span class="term">State</span> has been changed, and some new functions have been added.
+                The function <span class="term">State</span> has been changed, and some new functions have been added.<br>
```
Comment (RhenaudTheLukark) [inline Documentation CYK/documentation.html]: Why was this changed?

File: Documentation CYK/pages/api-functions-entity.html [hunk 4]
```diff
@@ -243,6 +243,11 @@ <h4>Player-specific variables and functions</h4><br>
                 call this function to update the UI accordingly.
             </p><br>
 
+            <p>
+                <br><span class="function"><span class="CYK2"></span> <span class="number"></span> <span class="term">hpAdd</span></span>
+                The number of HPs that will be added to your entity's HPs and max HPs if the fight is ended with a least one enemy "killed". 2 by default.
```
Comment (RhenaudTheLukark) [inline Documentation CYK/pages/api-functions-entity.html]: The amount of MaxHP that will be added to the entity's current MaxHP if at least one enemy was "killed" during this encounter. 2 by default.

File: Documentation CYK/pages/api-functions-entity.html [hunk 5]
```diff
@@ -374,6 +379,32 @@ <h4>Enemy-specific variables and functions</h4><br>
                 </ul>
             </p><br>
 
+            <p>
+                <span class="function"><span class="CYK2"></span> <span class="term">ChangeMercyPercent(<span class="number"></span> value, <span class="entity"></span> <span class="userdata">OR</span> <span class="string"></span> target(s) = self,
+                <span class="boolean"></span> showText = true, <span class="boolean"></span> playSound = true)</span></span>
+                Changes the mercy percentage of the <span class="term">target(s)</span> by the indicated <span class="term">value</span> and will show the yellow text and play the sound effect that goes with it if <span class="term">showText</span> and <span class="term">useSound</span> respectively are set to true.
```
Comment (RhenaudTheLukark) [inline Documentation CYK/pages/api-functions-entity.html]: The text's not necessarily yellow.

File: Documentation CYK/pages/api-functions-entity.html [hunk 6]
```diff
@@ -374,6 +379,32 @@ <h4>Enemy-specific variables and functions</h4><br>
...
+                    </ul>
+                    </li>
+                    <li><span class="term">showText</span>: If set to true, the text showing the <span class="term">value</span> will appear.</li>
+                    <li><span class="term">playSound</span>: If set to true, it'll play the sound effect used when the percentage is changed.</li>
+                </ul>
+            </p><br>
+
+            <p>
+                <span class="function"><span class="CYK2"></span> <span class="term">GetMercyPercent(<span class="entity"></span> target) returns <span class="number"></span></span></span>
+                Returns the percentage of mercy of the <span class="term">target</span>.<br>
```
Comment (RhenaudTheLukark) [inline Documentation CYK/pages/api-functions-entity.html]: mercy percentage

File: Documentation CYK/pages/variables.html [hunk 7]
```diff
@@ -660,8 +667,21 @@ <h4>Enemy object variables</h4>
             <p>
 				<br><span class="function"><span class="boolean"></span> <span class="term">canspare</span></span>
                 False by default. If you change this to true, your monster's name will turn yellow and it will be spareable.
+                <br><br>
+                <!--<span class="CYK2"></span> If <span class="term">chapter2</span> is set to true, <span class="term">canspare</span> will only act as a read-only variable returning true if the enemy can be spared and false otherwise.-->
+                <span class="CYK2"></span> If <span class="term">chapter2</span> is set to true, <span class="term">canspare</span> will indicates if the enemy will use the mercy counter and will be able to be spared through acts changing its percentage of mercy. If it's set to false, the counter will be deactivated and show a cross instead of a percentage.<br>Do note enemies can still be spared in that condition by calling <span class="term">Spare()</span>. True by default.
```
Comment (RhenaudTheLukark) [inline Documentation CYK/pages/variables.html]: Following the talk we've had, maybe move that feature over to a function named `DisableMercyBar(boolean)` or something.
You can also most likely remove the comments you added.

File: Documentation CYK/pages/variables.html [hunk 8]
```diff
@@ -660,8 +667,21 @@ <h4>Enemy object variables</h4>
...
 			</p>
 
+            <!--<p>
+                <br><span class="function"><span class="CYK2"></span><span class="boolean"></span> <span class="term">useMercyCounter</span></span>
+                If this variable is set to true, the enemy will use the mercy counter and will be able to be spared through acts changing its percentage of mercy. If it's set to false, the counter will be deactivated and show a cross instead of a percentage.<br>Do note enemies can still be spared in that condition by calling <span class="term">Spare()</span>. True by default.
+            </span>-->
+
+            <p>
+                <br><span class="function"><span class="CYK2"></span><span class="boolean"></span> <span class="term">useMercyCounter</span></span>
+                An alternative to <span class="term">canspare</span> when <span class="term">chapter2</span> is set to true.
```
Comment (RhenaudTheLukark) [inline Documentation CYK/pages/variables.html]: Please try to make so all functions start with a capital like most of CYK. `UseMercyCounter()` should be better.

File: Documentation CYK/pages/variables.html [hunk 9]
```diff
@@ -111,6 +111,13 @@ <h5>TIP: Lua has a very useful built-in string comparing function.</h5>
                 You can very easily check for <span class="term">if CYFversion &lt; "0.6.2.2" then</span>, <span class="term">if CYFversion &gt;= "0.6.1.2" then</span> and other combinations.
 			</p>
 
+            <p>
+                <br><span class="function"><span class="CYK2"></span><span class="boolean"></span> chapter2 [E/M/P]</span>
+                Activates all features from the Chapter 2 if it's true, only uses Chapter 1's otherwise.<br><br>
```
Comment (RhenaudTheLukark) [inline Documentation CYK/pages/variables.html]: "If it's set to true, the encounter will be played as if it was made in Deltarune's Chapter 2 with its various features enabled. Otherwise, CYK will look and play like an encounter made in Deltarune's Chapter 1."

**Review by RhenaudTheLukark — CHANGES_REQUESTED** [review]
Some stuff here and there, alongside the whole DisableMercyBar/Counter() thing.

File: Documentation CYK/documentation.html (hunk 3 above)
Comment (Simbel0) [reply Documentation CYK/documentation.html]: There was something here that I ended up removing but it looks like I missed this `<br>`
I guess I'll remove it at the next commit

File: Documentation CYK/pages/variables.html (hunk 8 above)
Comment (Simbel0) [reply Documentation CYK/pages/variables.html]: It's a variable

File: Documentation CYK/documentation.html (hunk 3 above)
Comment (RhenaudTheLukark) [reply Documentation CYK/documentation.html]: This wasn't addressed.

File: Documentation CYK/pages/api-functions-entity.html (hunk 4 above)
Comment (RhenaudTheLukark) [reply Documentation CYK/pages/api-functions-entity.html]: This wasn't addressed either.

File: Documentation CYK/pages/api-functions-entity.html (hunk 5 above)
Comment (RhenaudTheLukark) [reply Documentation CYK/pages/api-functions-entity.html]: Again.

File: Documentation CYK/pages/api-functions-entity.html (hunk 6 above)
Comment (RhenaudTheLukark) [reply Documentation CYK/pages/api-functions-entity.html]: Ditto.

File: Documentation CYK/pages/variables.html (hunk 9 above)
Comment (RhenaudTheLukark) [reply Documentation CYK/pages/variables.html]: Not addressed either.

File: Documentation CYK/pages/variables.html (hunk 7 above)
Comment (RhenaudTheLukark) [reply Documentation CYK/pages/variables.html]: This hasn't been changed either.

File: Lua/Libraries/CYK/AttackManager.lua [hunk 10]
```diff
@@ -341,7 +341,7 @@ return function(CYK)
             color = { 1, 1, 1 }
         end
 
-        CYK.UI.CreateHPChangeText(textValue, target, color)
+        CYK.UI.CreateChangeText(textValue, target, color)
```
Comment (RhenaudTheLukark) [inline Lua/Libraries/CYK/AttackManager.lua]: Why was the function's name changed here?

File: Lua/Libraries/CYK/EntityManager.lua [hunk 11]
```diff
@@ -1,5 +1,8 @@
 -- This library handles everything related to entities
 -- No wonder why it got its current name!
+
+
```
Comment (RhenaudTheLukark) [inline Lua/Libraries/CYK/EntityManager.lua]: Why those extra lines?

**Review by RhenaudTheLukark — CHANGES_REQUESTED** [review]
There are a few comments I made before and you didn't do anything about them. Please do something about them and I'll be happy to add that PR to CYK!

File: Lua/Libraries/CYK/AttackManager.lua (hunk 10 above)
Comment (Simbel0) [reply Lua/Libraries/CYK/AttackManager.lua]: Since it's now used for HPs AND Mercy Percentage, I changed its name to be more global.
It really was just changed because I'm a maniac-

File: Lua/Libraries/CYK/EntityManager.lua (hunk 11 above)
Comment (Simbel0) [reply Lua/Libraries/CYK/EntityManager.lua]: I think it's for when I tried to use the `getset` library in this file. I put the line here, deleted it later but forgot to remove the two lines around it

File: Lua/Libraries/CYK/AttackManager.lua (hunk 10 above)
Comment (RhenaudTheLukark) [reply Lua/Libraries/CYK/AttackManager.lua]: In that case, naming it CreateValueChangeText would be better I think.

File: Lua/Libraries/CYK/EntityManager.lua (hunk 11 above)
Comment (RhenaudTheLukark) [reply Lua/Libraries/CYK/EntityManager.lua]: Yeet 'em then, if they're relics of an old attempt, please.

File: Lua/Libraries/CYK/AttackManager.lua (hunk 10 above)
Comment (Simbel0) [reply Lua/Libraries/CYK/AttackManager.lua]: Got it, I'll change it

File: Lua/Libraries/CYK/EntityManager.lua (hunk 11 above)
Comment (Simbel0) [reply Lua/Libraries/CYK/EntityManager.lua]: I'm on it
